### PR TITLE
[ANCHOR-324] Add documentation for the AP architecture

### PR DESCRIPTION
**Pull request by Ifropc**

### CODE REVIEW

File: docs/anchoring-assets/anchor-platform/architecture.mdx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+This component is optional, and may be replaced by your own implementation.
+
+:::
+
+Stellar observer tracks Stellar chain and automatically detects user payments' Stellar transactions, updating corresponding SEP transactions statuses.
+
+#### Business Server
+
+Finally, Business server is a server that you (business) must implement, in order to fully support most of the SEPs, supported by the Platform. Business server reacts to some user events (such as requesting a quote for SEP-38), and pushes updates to the Platform for user notification (for example, completing a bank transfer)
```
Comment (JakeUrban) [inline docs/anchoring-assets/anchor-platform/architecture.mdx:42]:
```suggestion
Finally, The business server is a server that you (the business) must implement, in order to fully support most of the SEPs offered by the Anchor Platform. The business server responds to callback requests sent by the Anchor Platform (such as requesting a quote for SEP-38), and provides updates to the Anchor Platform when off-chain events occur (for example, completing a bank transfer).
```

File: docs/anchoring-assets/anchor-platform/architecture.mdx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## Future architecture
+
+:::caution
+
+This diagram is only served to show the planned architecture. Changes in the future versions may be different from the current diagram.
```
Comment (JakeUrban) [inline docs/anchoring-assets/anchor-platform/architecture.mdx:48]:
```suggestion
This diagram is only meant to show the planned architecture. Changes in future versions may be different from the diagrams provided.
```

File: docs/anchoring-assets/anchor-platform/architecture.mdx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+:::
+
+[![future anchor platform architecture](/assets/anchor-platform-architecture-2.png)](/assets/anchor-platform-architecture-2.png)
+
+### New Components
+
+#### Event Service
+
+Event service is a new component that will push events to both user and business via webhooks. Currently, some functionality of the event service is supported via [callback APIs][callback-api]
```
Comment (JakeUrban) [inline docs/anchoring-assets/anchor-platform/architecture.mdx:58]:
```suggestion
The event service is a new component that will push events to both the client and business via HTTP webhooks. Currently, some functionality of the event service is supported via [callback APIs][callback-api]
```

I don't think this second statement is accurate, and we might just want to remove it.

File: docs/anchoring-assets/anchor-platform/architecture.mdx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+#### Custody Service
```
Comment (JakeUrban) [inline docs/anchoring-assets/anchor-platform/architecture.mdx:60]:
```suggestion
#### Custody Server
```

File: docs/anchoring-assets/anchor-platform/architecture.mdx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+Custody service is a new component that allows business to connect to a custodial service (such as Fireblocks), in order to make payments and receive payments on the Stellar network. When used in self-custodial mode, this service is a next generation of the existing Stellar Observer.
```
Comment (JakeUrban) [inline docs/anchoring-assets/anchor-platform/architecture.mdx:62]:
```suggestion
The custody server is a new component that allows businesses to connect to a custodial service (such as Fireblocks), in order to send and receive payments on the Stellar network. When used in self-custodial mode, this service is a next generation of the existing Stellar Observer.
```

**Review by JakeUrban — APPROVED** [review]
Approved with some phrasing changes.

File: docs/anchoring-assets/anchor-platform/architecture.mdx (hunk 3 above)
Comment (Ifropc) [reply docs/anchoring-assets/anchor-platform/architecture.mdx:58]: I think I'll just move the callback API part to the SEP server description

**Review by briwylde08 — APPROVED** [review]

**Pull request merged**